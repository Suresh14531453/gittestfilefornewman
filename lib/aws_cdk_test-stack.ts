// import * as cdk from 'aws-cdk-lib';
// import { SecretValue } from 'aws-cdk-lib';
// import { SnsAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
// import { BuildSpec, LinuxBuildImage, PipelineProject } from 'aws-cdk-lib/aws-codebuild';
// import { Action, Artifact, Pipeline } from 'aws-cdk-lib/aws-codepipeline';
// import { CloudFormationCreateUpdateStackAction, CodeBuildAction, GitHubSourceAction } from 'aws-cdk-lib/aws-codepipeline-actions';
// import { Construct } from 'constructs';
// import { Topic } from "aws-cdk-lib/aws-sns";
// import { EmailSubscription } from "aws-cdk-lib/aws-sns-subscriptions";
// // import * as sqs from 'aws-cdk-lib/aws-sqs';
// // import { SnsTopic } from 'aws-cdk-lib/aws-events-targets';
// // import { EventField, RuleTargetInput } from 'aws-cdk-lib/aws-events';
// import * as cdk from 'aws-cdk-lib';
// import { SecretValue } from 'aws-cdk-lib';
// import { SnsAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
// import { BuildSpec, LinuxBuildImage, PipelineProject } from 'aws-cdk-lib/aws-codebuild';
// import { Action, Artifact, Pipeline } from 'aws-cdk-lib/aws-codepipeline';
// import { CloudFormationCreateUpdateStackAction, CodeBuildAction, GitHubSourceAction } from 'aws-cdk-lib/aws-codepipeline-actions';
// import { Construct } from 'constructs';


// export class AwsCdkTestStack extends cdk.Stack {
  
//   // private readonly pipelineNotificationsTopic: Topic;
//   constructor(scope: Construct, id: string, props?: cdk.StackProps) {
//     super(scope, id, props);
//     const pipeline=new Pipeline(this,"Pipeline",{
//       pipelineName: 'newmanpipeline',
//       crossAccountKeys: false,
//       restartExecutionOnUpdate: true,
//     })
    

//     const cdkSourceOutput=new Artifact("CDKSourceOutput")
//     pipeline.addStage({
//       stageName:"source",
//       actions:[
//         new GitHubSourceAction({
//           owner: "Suresh14531453",
//             repo: "gittestfilefornewman",
//             branch: "master",
//             actionName: "Pipeline_Source",
//             oauthToken: SecretValue.secretsManager("token_access"),
//             output: cdkSourceOutput
//         }
//         )
//       ]
//     })
//     const email = 'your.email@example.com';
//     const emailSubject = 'Build Stage Failed';
//     const emailText = 'Your pipeline has failed.';
//     const emailHtml = '<p>Your pipeline has failed.</p>';

//     const cdkBuildOutput=new Artifact("CdkBuildOutPut")
//     pipeline.addStage({
//       stageName:"build",
//       actions: [
//         new CodeBuildAction({

//           actionName: "CDK_Build",
//           input: cdkSourceOutput,
//           outputs: [cdkBuildOutput],
//           project: new PipelineProject(this, "CdkBuildProject", {
//             environment: {
//               buildImage: LinuxBuildImage.STANDARD_5_0,
//             },
//             buildSpec: BuildSpec.fromSourceFilename(
//               "build-specs/cdk-newman-build-spec.yml"
//             ),
//           }),
          
         
//         }),
//       ]
//     })
    
//     pipeline.addStage({
//       stageName: "Pipeline_Update",
//       actions: [
//         new CloudFormationCreateUpdateStackAction({
//           actionName: "Pipeline_Update",
//           stackName: "AwsCdkTestStack",
//           templatePath: cdkBuildOutput.atPath("AwsCdkTestStack.template.json"),
//           adminPermissions: true,
//         }),
//       ],
      
//     });
   
//   }
// }

import * as cdk from 'aws-cdk-lib';
import { SecretValue, Stage } from 'aws-cdk-lib';
import { BuildEnvironmentVariableType, BuildSpec, LinuxBuildImage, PipelineProject, Project } from 'aws-cdk-lib/aws-codebuild';
import { Artifact, IStage, Pipeline } from 'aws-cdk-lib/aws-codepipeline';
import { CloudFormationCreateUpdateStackAction, CodeBuildAction, CodeBuildActionType, GitHubSourceAction } from 'aws-cdk-lib/aws-codepipeline-actions';
import { EmailSubscription } from 'aws-cdk-lib/aws-sns-subscriptions';
import { Construct } from 'constructs';
import * as sns from '@aws-cdk/aws-sns';
import { Topic } from "aws-cdk-lib/aws-sns";
import { SnsTopic } from 'aws-cdk-lib/aws-events-targets';
import { EventField, RuleTargetInput } from 'aws-cdk-lib/aws-events';

export class AwsCdkTestStack extends cdk.Stack {
  private readonly pipelineNotificationsTopic: Topic;
  private readonly buildFailureTopic:Topic;

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    
    this.pipelineNotificationsTopic = new Topic(
      this,
      "PipelineNotificationsTopic",
      {
        topicName: "PipelineNotifications",
      }
    );
    this.pipelineNotificationsTopic.addSubscription(
      new EmailSubscription("sureshsahu1453@gmail.com")
    );    
    const pipeline = new Pipeline(this, "Pipeline", {
      pipelineName: 'newmanpipeline',
      crossAccountKeys: false,
      restartExecutionOnUpdate: true,
    })

    const cdkSourceOutput = new Artifact("CDKSourceOutput")
    pipeline.addStage({
      stageName: "source",
      actions: [
        new GitHubSourceAction({
          owner: "Suresh14531453",
          repo: "gittestfilefornewman",
          branch: "master",
          actionName: "Pipeline_Source",
          oauthToken: SecretValue.secretsManager("token_access"),
          output: cdkSourceOutput
        }
        )
      ]
    })

    const cdkBuildOutput = new Artifact("CdkBuildOutPut")
    // pipeline.addStage({
    //   stageName:"build",
    //   actions: [
    //     new CodeBuildAction({

    //       actionName: "CDK_Build",
    //       input: cdkSourceOutput,
    //       outputs: [cdkBuildOutput],
    //       project: new PipelineProject(this, "CdkBuildProject", {
    //         environment: {
    //           buildImage: LinuxBuildImage.STANDARD_5_0,
    //         },
    //         buildSpec: BuildSpec.fromSourceFilename(
    //           "build-specs/cdk-newman-build-spec.yml"
    //         ),
    //       }),
    //     }),
    //   ],
    //   // onFail: new SnsAction({
    //   //       actionName: 'NotifyFailure',
    //   //       topic: snsTopic,
    //   //       message: pipeline.ArtifactPath.fromSourcePath('Build failed')
    //   //     })
    //   new SnsAction({
    //     actionName: "NotifyFailure",
    //     topic: snsTopic,
    //     message: "Build failed",
    //     runOrder: 2,
    //   }),
      
    // })
    const buildStage=pipeline.addStage({
      stageName: "build",
      actions: [
        new CodeBuildAction({
          actionName: "CDK_Build",
          input: cdkSourceOutput,
          outputs: [cdkBuildOutput],
          project: new PipelineProject(this, "CdkBuildProject", {
            environment: {
              buildImage: LinuxBuildImage.STANDARD_5_0,
            },
            buildSpec: BuildSpec.fromSourceFilename(
              "build-specs/cdk-newman-build-spec.yml"
            ),
          }),
        }),
       
      ],
    });
    buildStage.onStateChange(
      "FAILED",
      new SnsTopic(this.pipelineNotificationsTopic, {
        message: RuleTargetInput.fromText(
          `Build Test Failed By Syed. See details here: ${EventField.fromPath(
            "$.detail.execution-result.external-execution-url"
          )}`
        ),
      }),
      {
        ruleName: "Failed",
        eventPattern: {
          detail: {
            state: ["FAILED"],
          },
        },
        description: "Integration test has failed by syed",
      }
    );
    buildStage.onStateChange(
      "Succeeded",
      new SnsTopic(this.pipelineNotificationsTopic, {
        message: RuleTargetInput.fromText(
          `Build Test Failed By Syed. See details here: ${EventField.fromPath(
            "$.detail.execution-result.external-execution-url"
          )}`
        ),
      }),
      {
        ruleName: "SUCCEDED",
        eventPattern: {
          detail: {
            state: ["SUCCEEDED"],
          },
        },
        description: "Integration test has SUCCESS by SURESH",
      }
    );
   

  }
}
//////////////////////////////
