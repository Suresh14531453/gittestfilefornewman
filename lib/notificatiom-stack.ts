import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
export class AwsCdkTestStack extends cdk.Stack {
  
    // private readonly pipelineNotificationsTopic: Topic;
    constructor(scope: Construct, id: string, props?: cdk.StackProps) {
      super(scope, id, props);
    }

}   
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               