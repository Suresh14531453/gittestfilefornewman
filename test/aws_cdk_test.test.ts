 import * as cdk from 'aws-cdk-lib';
 import * as AwsCdkTest from '../lib/aws_cdk_test-stack';
 import { Match, Template } from "aws-cdk-lib/assertions";
// example test. To run these tests, uncomment this file along with the
// example resource in lib/aws_cdk_test-stack.ts
test('SQS Queue Created', () => {
  const app = new cdk.App();
//     // WHEN
  const stack = new AwsCdkTest.AwsCdkTestStack(app, 'MyTestStack');
//     // THEN
  const template = Template.fromStack(stack);
  // expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
  // template.hasResourceProperties('AWS::SQS::Queue', {
  //   VisibilityTimeout: 300
  // });
});
