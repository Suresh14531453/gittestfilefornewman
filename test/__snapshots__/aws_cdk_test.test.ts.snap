// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SQS Queue Created 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CdkBuildProject9382C38D": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "CdkBuildProjectRoleE0B6FEB0",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "build-specs/cdk-newman-build-spec.yml",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "CdkBuildProjectRoleDefaultPolicy3C7ECB00": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "CdkBuildProject9382C38D",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkBuildProjectRoleDefaultPolicy3C7ECB00",
        "Roles": [
          {
            "Ref": "CdkBuildProjectRoleE0B6FEB0",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkBuildProjectRoleE0B6FEB0": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineArtifactsBucket22248F97": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "PipelineArtifactsBucketPolicyD4F9712A": {
      "Properties": {
        "Bucket": {
          "Ref": "PipelineArtifactsBucket22248F97",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "PipelineC660917D": {
      "DependsOn": [
        "PipelineRoleDefaultPolicyC7A05455",
        "PipelineRoleD68726F7",
      ],
      "Properties": {
        "ArtifactStore": {
          "Location": {
            "Ref": "PipelineArtifactsBucket22248F97",
          },
          "Type": "S3",
        },
        "Name": "newmanpipeline",
        "RestartExecutionOnUpdate": true,
        "RoleArn": {
          "Fn::GetAtt": [
            "PipelineRoleD68726F7",
            "Arn",
          ],
        },
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:token_access:SecretString:::}}",
                  "Owner": "Suresh14531453",
                  "PollForSourceChanges": false,
                  "Repo": "gittestfilefornewman",
                },
                "Name": "Pipeline_Source",
                "OutputArtifacts": [
                  {
                    "Name": "CDKSourceOutput",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "source",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "ProjectName": {
                    "Ref": "CdkBuildProject9382C38D",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "CDKSourceOutput",
                  },
                ],
                "Name": "CDK_Build",
                "OutputArtifacts": [
                  {
                    "Name": "CdkBuildOutPut",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "PipelinebuildCDKBuildCodePipelineActionRole83F3AB81",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "build",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "PipelineNotificationsTopic827A419F": {
      "Properties": {
        "TopicName": "PipelineNotifications",
      },
      "Type": "AWS::SNS::Topic",
    },
    "PipelineNotificationsTopicPolicy68C85C82": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sns:Publish",
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
              "Resource": {
                "Ref": "PipelineNotificationsTopic827A419F",
              },
              "Sid": "0",
            },
          ],
          "Version": "2012-10-17",
        },
        "Topics": [
          {
            "Ref": "PipelineNotificationsTopic827A419F",
          },
        ],
      },
      "Type": "AWS::SNS::TopicPolicy",
    },
    "PipelineNotificationsTopicsureshsahu1453gmailcomF33BACF2": {
      "Properties": {
        "Endpoint": "sureshsahu1453@gmail.com",
        "Protocol": "email",
        "TopicArn": {
          "Ref": "PipelineNotificationsTopic827A419F",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "PipelineRoleD68726F7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineRoleDefaultPolicyC7A05455": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelinebuildCDKBuildCodePipelineActionRole83F3AB81",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineRoleDefaultPolicyC7A05455",
        "Roles": [
          {
            "Ref": "PipelineRoleD68726F7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelinebuildCDKBuildCodePipelineActionRole83F3AB81": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelinebuildCDKBuildCodePipelineActionRoleDefaultPolicy26FE4F83": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkBuildProject9382C38D",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelinebuildCDKBuildCodePipelineActionRoleDefaultPolicy26FE4F83",
        "Roles": [
          {
            "Ref": "PipelinebuildCDKBuildCodePipelineActionRole83F3AB81",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelinebuildFAILEDABA22781": {
      "Properties": {
        "Description": "Build Test Failed",
        "EventPattern": {
          "detail": {
            "stage": [
              "build",
            ],
            "state": [
              "FAILED",
            ],
          },
          "detail-type": [
            "CodePipeline Stage Execution State Change",
          ],
          "resources": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":codepipeline:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":",
                  {
                    "Ref": "PipelineC660917D",
                  },
                ],
              ],
            },
          ],
          "source": [
            "aws.codepipeline",
          ],
        },
        "Name": "Failed",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Ref": "PipelineNotificationsTopic827A419F",
            },
            "Id": "Target0",
            "Input": ""Build Test Failed"",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "PipelinebuildSUCCEEDEDF37AFF53": {
      "Properties": {
        "Description": "Build Test Successful",
        "EventPattern": {
          "detail": {
            "stage": [
              "build",
            ],
            "state": [
              "SUCCEEDED",
            ],
          },
          "detail-type": [
            "CodePipeline Stage Execution State Change",
          ],
          "resources": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":codepipeline:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":",
                  {
                    "Ref": "PipelineC660917D",
                  },
                ],
              ],
            },
          ],
          "source": [
            "aws.codepipeline",
          ],
        },
        "Name": "Success",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Ref": "PipelineNotificationsTopic827A419F",
            },
            "Id": "Target0",
            "Input": ""Build Test Successful."",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "PipelinesourcePipelineSourceWebhookResource4E958C6E": {
      "Properties": {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": {
          "SecretToken": "{{resolve:secretsmanager:token_access:SecretString:::}}",
        },
        "Filters": [
          {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "Pipeline_Source",
        "TargetPipeline": {
          "Ref": "PipelineC660917D",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
